src/module/actor/data-model-character.js

```diff
@@ -0,0 +1,204 @@
+import OseDataModelCharacterAC from "./dataModelClasses/OseDataModelCharacterAC";
+import OseDataModelCharacterEncumbrance from "./dataModelClasses/OseDataModelCharacterEncumbrance";
+import OseDataModelCharacterMove from "./dataModelClasses/OseDataModelCharacterMove";
+import OseDataModelCharacterScores from "./dataModelClasses/OseDataModelCharacterScores";
+import OseDataModelCharacterSpells from "./dataModelClasses/OseDataModelCharacterSpells";
+
+const getItemsOfActorOfType = (actor, filterType, filterFn = null) =>
+  actor.items
+    .filter(({type}) => type === filterType)
+    .filter(filterFn ? filterFn : () => true);
+
+export default class OseDataModelCharacter extends foundry.abstract.DataModel {
+  prepareDerivedData() {
+    this.scores = new OseDataModelCharacterScores(this.scores);
+
+    this.encumbrance = new OseDataModelCharacterEncumbrance(
+      game.settings.get(game.system.id, "encumbranceOption"),
+      this.encumbrance.max,
+      [...this.parent.items],
+      game.settings.get(game.system.id, "significantTreasure"),
+    );
+
+    this.movement = new OseDataModelCharacterMove(
+      this.encumbrance,
+      this.config.movementAuto,
+      this.movement.base
+    );
+
+    this.ac = new OseDataModelCharacterAC(
+      false,
+      [...getItemsOfActorOfType(this.parent, 'armor', (a) => a.system.equipped)],
+      this.scores.dex.mod,
+      this.ac.mod,
+    );
+
+    this.aac = new OseDataModelCharacterAC(
+      true,
+      getItemsOfActorOfType(this.parent, 'armor', (a) => a.system.equipped),
+      this.scores.dex.mod,
+      this.aac.mod,
+    );
+
+    this.spells = new OseDataModelCharacterSpells(
+      this.spells,
+      this.#spellList
+    )
+  }
+
+  // @TODO set up schema
+  // @todo define schema
+  static defineSchema() {
+    const { SchemaField, StringField, NumberField, BooleanField, ArrayField, ObjectField } = foundry.data.fields;
+
+    return {
+      spells: new ObjectField(),
+      scores: new ObjectField(),
+      details: new ObjectField(),
+      ac: new ObjectField(),
+      aac: new ObjectField(),
+      encumbrance: new ObjectField(),
+      movement: new ObjectField(),
+      config: new ObjectField(),
+      initiative: new ObjectField(),
+      hp: new ObjectField(),
+      thac0: new ObjectField(),
+      initiative: new ObjectField(),
+      init: new NumberField({readonly: true}),
+      rangedMod: new NumberField({readonly: true}),
+      meleeMod: new NumberField({readonly: true}),
+      usesAscendingAC: new BooleanField({readonly: true}),
+    };
+  }
+  
+  get usesAscendingAC() {
+    return game.settings.get(game.system.id, 'ascendingAC');
+  }
+  
+  get meleeMod() {
+    const ascendingAcMod = game.settings.get(game.system.id, 'ascendingAC')
```

Comment:
This applies an ascending AC mod if one exists; a few lines down, it adds this value onto the returned value.

This is 0 for descending AC, since descending AC has THAC0 instead.

Reply:
Oh! I can't do that, because we're still setting `ac` and `aac`, which we're using in the existing character sheet. I don't know how far-reaching the pattern of using both AC and AAC is, so I don't want to break anything.

Once we get into the new character sheet, we should be able to drop the two separate AC props and unify on `actor.system.ac`

Reply:
Wait. I'm a dummy. I get what you're saying now, hahaha.

I'll fix this.